src/Repository/index.js

```diff
@@ -140,41 +150,41 @@ class Repository {
         await Promise.all(createOrUpdateFilesMultiple)
     }
 
-    async createPullRequest({ title, body, branchName, defaultBranch }) {
+    async createPullRequest({ title, body, branchName }) {
         const result = await this.github.pulls.create({
             owner: this.owner,
             repo: this.repo,
             title,
             body,
             head: branchName,
-            base: defaultBranch,
+            base: this.defaultBranch,
             maintainer_can_modify: true,
         })
         return result.data.html_url
     }
 
-    async createPullRequestFromFiles({
-        title,
-        body,
-        filesByPath,
-        branchName,
-        defaultBranch,
-    }) {
-        await this.createBranch({ branchName, defaultBranch })
+    async createPullRequestFromFiles({ title, body, filesByPath, branchName }) {
+        if (this.basedBranch === this.defaultBranch)
+            await this.createBranch(branchName)
 
         await this.createOrUpdateFiles({
             filesByPath,
             branchName,
         })
 
-        const pullRequestURL = await this.createPullRequest({
-            title,
-            body,
-            branchName,
-            defaultBranch,
-        })
+        try {
+            const pullRequestURL = await this.createPullRequest({
+                title,
+                body,
+                branchName,
+            })
+            return pullRequestURL
+        } catch (error) {
+            if (error.status === 422)
+                throw new AllContributorBotError('Pull request is already open')
```

Comment:
Can we add a link?
`I've updated the [pull request](INSERT_LINK_HERE) you have open.`

I don't think this should error too, could we handle this as a reply case here: (https://github.com/all-contributors/all-contributors-bot/blob/master/src/processIssueComment.js#L65-L67) e.g. can throw the error, but catch it here